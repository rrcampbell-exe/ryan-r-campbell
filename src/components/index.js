import BlogTags from './Blog/BlogTags'
import BlogTeaser from './Blog/BlogTeaser'
import Books from './Books/Books'
import Builds from './Builds/Builds'
import Contact from './Contact/Contact'
import Footer from './Footer/Footer'
import Header from './Header/Header'
import PageWrapper from './PageWrapper/PageWrapper'
import SubNav from './SubNav/SubNav'
import Reviews from './Reviews/Reviews'
import NotFound from './NotFound/NotFound'
import NotificationBanner from './NotificationBanner/NotificationBanner'
import Loading from './Loading/Loading'
import Quote from './Quote/Quote'
import TextAndImage from './TextAndImage/TextAndImage'
import HamburgerMenu from './HamburgerMenu/HamburgerMenu'
import PopText from './PopText/PopText'
import PodcastTeaser from './Podcast/PodcastTeaser'
import Boasts from './Tech/Boasts'
import SkillsCloud from './Tech/SkillsCloud'
import BlogFooter from './Blog/BlogFooter'
import SubstackRedirect from './SubstackRedirect/SubstackRedirect'
import { CodeSpan, CodeBlock } from './CodeBlock/CodeBlock'

export { BlogTeaser, Books, Builds, Footer, Header, PageWrapper, SubNav, Contact, Reviews, NotFound, NotificationBanner, Loading, Quote, TextAndImage, BlogTags, HamburgerMenu, PopText, PodcastTeaser, Boasts, SkillsCloud, BlogFooter, CodeSpan, CodeBlock, SubstackRedirect }
