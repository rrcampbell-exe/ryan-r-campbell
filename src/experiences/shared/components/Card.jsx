import React from 'react'
import styled from 'styled-components'

// TODO: size should flex based on number of cards
// TODO: need to set up theme colors

const StyledDiv = styled.div`
  height: 16rem;
  width: 16rem;
  border: 1px solid #242424;
  border-radius: 4px;
  padding: 1rem;
  margin: 1rem;
`

const Card = ({ children }) => {
  return (
    <StyledDiv>
      {children}
    </StyledDiv>
  )
}

export default Card
