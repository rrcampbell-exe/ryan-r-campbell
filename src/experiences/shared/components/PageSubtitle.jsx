import React from 'react'
import styled from 'styled-components'

const StyledH2 = styled.h2`
  margin: 0 auto;
  padding: 0;
`

const PageSubtitle = ({ subtitle }) => {
  return (
    <StyledH2>
      {subtitle}
    </StyledH2>
  )
}

export default PageSubtitle
