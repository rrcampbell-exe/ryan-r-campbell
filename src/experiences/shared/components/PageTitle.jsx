import React from 'react'
import styled from 'styled-components'

const StyledH1 = styled.h1`
  margin: 0 auto;
  padding: 0;
`

const PageTitle = ({ title }) => {
  return (
    <StyledH1>
      {title}
    </StyledH1>
  )
}

export default PageTitle
