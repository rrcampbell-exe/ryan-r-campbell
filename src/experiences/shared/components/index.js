import BlogPost from './Blog/BlogPost'
import BlogTeaser from './Blog/BlogTeaser'
import Books from './Books/Books'
import Builds from './Builds/Builds'
import Footer from './Footer/Footer'
import Header from './Header/Header'
import PageWrapper from './PageWrapper/PageWrapper'
import SubNav from './SubNav/SubNav'
import Contact from './Contact/Contact'
import Reviews from './Reviews/Reviews'
import NotFound from './NotFound/NotFound'
import NotificationBanner from './NotificationBanner/NotificationBanner'

export { BlogPost, BlogTeaser, Books, Builds, Footer, Header, PageWrapper, SubNav, Contact, Reviews, NotFound, NotificationBanner }
