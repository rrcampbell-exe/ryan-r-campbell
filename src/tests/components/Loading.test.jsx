import { describe, expect, test } from 'vitest'
import { render } from '@testing-library/react'
import ContextAndRouterProvider from '../test-utils'
import { Loading } from '../../components'

describe('Loading', () => {
  test('default snapshot', () => {
    const loading = render(
      <ContextAndRouterProvider>
        <Loading />
      </ContextAndRouterProvider>
    )
    expect(loading).toMatchSnapshot()
  })
  test('should render loading text', () => {
    const { getByText } = render(
      <ContextAndRouterProvider>
        <Loading />
      </ContextAndRouterProvider>
    )

    const loadingText = getByText('loading...')
    expect(loadingText).toBeInTheDocument()
  })
  test('should render MicrophoneSVG', () => {
    const { container } = render(
      <ContextAndRouterProvider>
        <Loading />
      </ContextAndRouterProvider>
    )

    const microphoneSVG = container.querySelector('#microphone-svg')
    expect(microphoneSVG).toBeInTheDocument()
  })
})
