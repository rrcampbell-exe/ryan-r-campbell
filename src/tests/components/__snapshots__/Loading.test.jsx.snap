// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loading > default snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Loading"
      >
        <div
          class="loader-content"
        >
          <svg
            fill="none"
            height="45px"
            id="microphone-svg"
            viewBox="0 0 24 24"
            width="45px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 10V12C19 15.866 15.866 19 12 19M5 10V12C5 15.866 8.13401 19 12 19M12 19V22M8 22H16M15 6H13M15 10H13M12 15C10.3431 15 9 13.6569 9 12V5C9 3.34315 10.3431 2 12 2C13.6569 2 15 3.34315 15 5V12C15 13.6569 13.6569 15 12 15Z"
              stroke="#000000"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <p>
            loading...
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Loading"
    >
      <div
        class="loader-content"
      >
        <svg
          fill="none"
          height="45px"
          id="microphone-svg"
          viewBox="0 0 24 24"
          width="45px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 10V12C19 15.866 15.866 19 12 19M5 10V12C5 15.866 8.13401 19 12 19M12 19V22M8 22H16M15 6H13M15 10H13M12 15C10.3431 15 9 13.6569 9 12V5C9 3.34315 10.3431 2 12 2C13.6569 2 15 3.34315 15 5V12C15 13.6569 13.6569 15 12 15Z"
            stroke="#000000"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
        <p>
          loading...
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
